### ISSUE TITLE
DEP-3 — Secondary integrations: OpenMM, Symmetrix, libmace, and the full-basis weights export

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1595]
**Depends on:** DEP-1 (#1583), BKD-3 (#1567), FM-2 (#1587) · **Blocks:** —

**Context:** Deployment (DEP-2 (#1594)) ships the LAMMPS/MLIAP path and a shared loader,
`load_deployed(bundle_path, *, device="cpu") -> DeployedModel` in `mace_torch.deploy`, that
returns the built eager model plus its manifest and exposes `.compute(graph) -> MACEOutputs`.
Three secondary integrations named as v1.0 must-haves still hang off that loader and off the
foundation artifacts, and none is on the critical path — so this ticket handles them **in an
explicit internal de-scope order** (OpenMM → Symmetrix sweep → libmace): if capacity runs out,
cut from the tail, and record the cut rather than dropping it silently. None of the three has
any in-tree presence on develop — there is no `tests/integrations/openmm`, no
`tests/integrations/libmace`, and no Symmetrix benchmark row — so each is new surface, and
each is created under the two-tier discipline `tests/integrations/README.md` already fixes:
a contract tier that needs no external binary, plus a `bin_<x>`-marked real tier registered in
`pyproject.toml` and `CAPABILITY_PROBES` and run from `nightly.yaml`.

The one non-obvious cost belongs here, named up front rather than discovered mid-integration:
**Symmetrix and libmace reimplement the *original full-basis* symmetric contraction and read
weights**, but every v1 artifact stores the **reduced-basis** canonical weights (the layout
CORE-4 (#1558) pins; FM-2 (#1587) publishes the converted foundation models in it). A v1 → Symmetrix / libmace
path therefore cannot hand over the checkpoint tensors as-is. Two ways out exist; this ticket
picks the first and states its price:

- **(a)** emit a one-way **full-basis weights export** for the external consumer. `reduced → full`
  is exact but under-determined — infinitely many full-basis weight vectors reproduce the same
  function; the export picks the **minimum-norm representative** among the gauge directions.
  Measured exact at fp64: residual `2.0e-13` on MACE-MP-0 small (13 → 28 paths) and `1.7e-13`
  on MACE-OFF23 medium (29 → 86 paths). Predictions are identical.
- **(b)** teach Symmetrix the reduced basis (a change in an external codebase MACE does not own).

Path (a) is the shipped route; (b) is recorded as the alternative.

**Interface & constraints:**

1. **Full-basis weights export** — a new subcommand under CLI-1 (#1579)'s `mace export`:

   ```
   mace export full-basis-weights <bundle>  ->  <name>-full-basis.safetensors
                                              +  <name>-full-basis.json
   ```

   The safetensors payload holds one flat `[Z, A, mul]` full-basis contraction tensor
   per product block, keyed by block index, in the full-basis path order and per-path
   normalization derived from the canonical reduced-basis convention. The JSON sidecar
   records the irreps configuration, correlation order, the ordered path list, the
   element ordering and a schema version — enough for an external reimplementation to
   map paths without reading MACE source. Agree the key names and path ordering with
   the Symmetrix and libmace maintainers before implementing, and pin the exact commit
   of each tool the benchmark sweep and verification case run against — the same
   pin-by-commit discipline `requirements/polar.txt` and `requirements/les.txt` already
   apply to the other external dependencies.

   It is an **export**: a derived, one-way artifact for external reimplementations, **not** a
   checkpoint. It is *not* loadable as a v1 model. It is distinct from the checkpoint converter
   `mace model convert --to-basis` (CLI-1 (#1579)), which does the lossless `full → reduced` **downgrade**
   and **refuses `reduced → full`** because that direction is not unique. The distinction is the
   whole point: the export is allowed to pick a representative precisely because its consumer only
   evaluates the function, never resumes training on the weights.
2. **Exactness is golden-gated, not weight-gated.** The correctness statement is *predictions
   match*, never *weights match*: the full-basis weights are a gauge representative, so gate on the
   committed E/F/stress goldens at fp64 tolerance, never on weight equality. `full → reduced`
   applied to the export must return the original reduced weights exactly (round-trip closes the
   loop and proves the representative is valid).
3. **OpenMM** consumes the neutral bundle through the **same `load_deployed`** DEP-2 (#1594) defines — no
   new loader, no new serialization format. It is the OpenMM direct-PyTorch (NNPOps-style)
   interface: build the eager model once from the bundle, evaluate energy+forces per step. It
   shares the load half of the deployment class with ASE (DEP-1 (#1583) wraps the same loader) and MLIAP,
   so the "one backend selection loads it everywhere" goal is realized by wiring, not by a second
   code path. There is an existing `docs/guide/openmm.rst` on the `docs` branch describing the
   v0.3.x route; REL-3 (#1605) rewrites it against this one.
4. **Symmetrix** is the perf gold standard for older (full-basis) models. The deliverable is a
   **benchmark sweep across model sizes and atom counts** comparing the v1 eager(+`torch.compile`,
   and the BKD-3 (#1567) accelerated backends) engine against Symmetrix on the *same* full-basis
   foundation model — the full-basis export is what feeds Symmetrix the weights. It lands as a
   row in the nightly `benchmarks` job (`nightly.yaml`, which already runs `tests/benchmarks` with
   `--benchmark-json`), not as a correctness gate.
5. **libmace** (the Fortran reference interface) is maintained as a **supported integration path**
   at a defined level: the full-basis export loads into libmace and reproduces v1 E/F within a
   documented tolerance (a shipped **verification case** in a new `tests/integrations/libmace/`
   with its own `bin_libmace` marker and probe), and a **support-status doc** states what
   is verified vs benchmark-only for each of the three integrations and records the reduced-vs-full
   basis cost from Context.

**Task:**
1. Add `mace export full-basis-weights` (CLI-1 (#1579) `mace export` subtree): read the bundle's
   reduced-basis contraction weights, compute the minimum-norm `reduced → full` representative per
   element/block, write the full-basis artifact for external reimplementations. Refuse to treat it
   as a checkpoint (it is export-only).
2. Golden-gate the export: exported full-basis weights reproduce the v1 model's E/F/stress at fp64
   tolerance on the tiny anchors; `full → reduced` of the export returns the original reduced
   weights exactly.
3. **OpenMM:** wire the direct-PyTorch interface onto `load_deployed`; a contract test loads a v1
   bundle and matches the DEP-1 (#1583) ASE calculator's E/F on a shared structure at golden tolerance.
4. **Symmetrix:** build the benchmark sweep (model sizes × atom counts), feeding Symmetrix the
   full-basis export at a pinned commit; commit it as a nightly benchmark row reporting v1 vs Symmetrix.
5. **libmace:** create `tests/integrations/libmace/` with a `bin_libmace` marker + probe, ship the
   verification case (full-basis export → libmace → E/F vs v1 within documented tolerance), and write
   the support-status doc covering all three integrations and the basis cost. The doc is the shared
   home for integration support levels, so DEP-4 (#1596) adds the torch-sim row to it — leave the table
   extensible rather than hardcoding three entries.

**Out of scope:** the LAMMPS/MLIAP path, `load_deployed`, the deploy bundle/manifest and the
trace emergency-exit (all DEP-2 (#1594)); the checkpoint converter `mace model convert --to-basis` and its
`reduced → full` refusal (CLI-1 (#1579) — cross-checked here, not built); publishing/registry of the
foundation artifacts (FM-2 (#1587)/FM-3 (#1588)); JAX deployment (there is none).

**De-scope order (binding):** if Phase 5 capacity runs short, cut in this order — OpenMM →
Symmetrix sweep → libmace — and record the cut explicitly (it is a recorded deviation from a
must-have, never a silent drop). The full-basis export underpins Symmetrix and libmace, so it is
the last of the four to go.

**Acceptance criteria:**
- [ ] `mace export full-basis-weights <bundle>` emits full-basis contraction weights; the exported
      weights reproduce the reduced-basis v1 model's E/F/stress at the fp64 golden tolerance on the
      tiny anchors (function-preserving, gated on predictions, not weight equality).
- [ ] The export is one-way: the artifact is not loadable as a v1 checkpoint, and `full → reduced`
      of the export returns the original reduced weights exactly (round-trip).
- [ ] `mace model convert --to-basis reduced→full` still refuses (unchanged CLI-1 (#1579) behaviour,
      cross-checked so the two `reduced→full` paths do not blur).
- [ ] OpenMM's direct-PyTorch interface loads a v1 bundle via `load_deployed` and matches the DEP-1 (#1583)
      ASE calculator's E/F on a shared structure at golden tolerance.
- [ ] A Symmetrix benchmark sweep across model sizes and atom counts runs on the same full-basis
      foundation model, at a pinned Symmetrix commit, and is committed as a nightly benchmark row.
- [ ] A libmace verification case reproduces v1 E/F from the full-basis export within its documented
      tolerance; `bin_libmace` is registered in `pyproject.toml` and `CAPABILITY_PROBES`, skips
      locally without the binary, and fails under `MACE_REQUIRE_CAPS`.
- [ ] A support-status doc records verified-vs-benchmark level and the basis cost for
      OpenMM / Symmetrix / libmace, in a table DEP-4 (#1596) can extend.

**Verify:**
```bash
# CPU-runnable correctness gate (export round-trip + parity + refusal cross-check)
python -m pytest packages/mace-torch/tests/deploy/test_full_basis_export.py -v
python -m pytest tests/parity -m "not slow" -k full_basis
# External integrations run in nightly (need OpenMM / Symmetrix / a libmace build):
python -m pytest packages/mace-torch/tests/deploy/test_openmm_integration.py -v   # nightly / opt-in
python -m pytest tests/benchmarks -k symmetrix                                    # nightly sweep
python -m pytest tests/integrations/libmace -m bin_libmace -v                     # nightly, needs libmace
```
**Review focus:** that the `reduced → full` export is the minimum-norm representative and stays
**export-only** (never a loadable checkpoint, never confused with the CLI-1 (#1579) converter's refused
upgrade), and that exactness is gated on E/F/stress goldens rather than weight equality; the basis
cost and each integration's support level must be truthfully stated in the support-status doc.

---
